### PR TITLE
FLUID-5795: Standard builds will publish a version control tag

### PR DESCRIPTION
**Pull request by jobara**
https://issues.fluidproject.org/browse/FLUID-5795

also added a fix for
https://issues.fluidproject.org/browse/FLUID-5798

### CODE REVIEW

File: publish.js [hunk 1]
```diff
@@ -54,6 +54,21 @@ publish.getCLIOpts = function () {
...
+
+/**
+ * Returns the contents of a package.json file as JSON object
+ *
+ * @param moduleRoot {String} - the path to the root where the package.json is
+ *                              located. Will use process.cwd() by default.
+ * @returns {Object} - returns the contents of the package.json file as a JSON
+ *                     object.
+ */
+publish.getPkg = function (moduleRoot) {
```
Comment (amb26) [inline publish.js]: With this utility you may as well rewrite line 18 of this file to 
var pkg = publish.getPkg(__dirname)
and eliminate pkgPath
(together with a suitable reordering of the definitions in the file)

File: publish.js [hunk 2]
```diff
@@ -125,6 +138,25 @@ publish.checkChanges = function (options) {
...
+ * Calls publish.execSync with a
+ * If it is a test run, the command will be output to the console.
+ *
+ * @param cmdTemplate {String} - A string template of the command to execute.
+ *                               Can provide tokens in the form ${tokenName}
+ * @param values {Object} - the tokens and their replacement.
+                            e.g. {tokenName: "value to insert"}
+ * @param isTest {Boolean} - indicates if this is a test run or not
+ */
+ publish.execSyncFromTemplate = function (cmdTemplate, cmdValues, isTest) {
```
Comment (amb26) [inline publish.js]: Linting problem here

**Comment by amb26** [comment]
I imagine that the newline entres the "revision" string at line 184 here - https://github.com/jobara/fluid-publish/blob/FLUID-5795/publish.js#L184 - looks like String.trim() has broad enough support to be used in node for this

**Comment by jobara** [comment]
@amb26 ready for more review. @colinbdclark This should also address the issue you found with the dist-tag not working.

**Pull request merged**